### PR TITLE
Update phpstan/phpstan related dependencies to ^2.0"

### PR DESCRIPTION
**Pull request by Boegie**
Naive first stab

### CODE REVIEW

**Comment by mglaman** [comment]
I'll probably want to open a 2.x branch as well. Thanks for trying at this.

**Comment by Boegie** [comment]
No worries, I'll chip away at this for an hour 2 from now.

**Comment by Boegie** [comment]
Ok, so this was more of a grind than actually Big Brain stuff.

I would recommend:

- Going over the identifiers that were (forcefully) added. There  are a lot of them and they might not be consistent.
- The changes in `.github/workflows/php.yml` can be reverted when drupal core includes `phpstan/phpstan:^2.0`
- Replacing all `@phpstan-ignore next-line` nuke-em-all lines with the more targeted `@phpstan-ignore [identifier]` in a separate MR. I found some "hidden sins" whilst creating this MR.

File: src/Rules/Classes/PluginManagerInspectionRule.php [hunk 1]
```diff
@@ -78,9 +78,9 @@ public function processNode(Node $node, Scope $scope): array
             $errors[] = RuleErrorBuilder::message(
                 'Plugin managers should call alterInfo to allow plugin definitions to be altered.'
             )
-                ->identifier('plugin.manager.alterInfoMissing')
                 ->tip('For example, to invoke hook_mymodule_data_alter() call alterInfo with "mymodule_data".')
                 ->line($node->getStartLine())
+                ->identifier('pluginManagerInspection.alterInfoMissing')
                 ->build();
```
Comment (mglaman) [inline src/Rules/Classes/PluginManagerInspectionRule.php:84]: I didn't review the commit log, but I'm guessing something about the format of the error identifier made PHPStan mad?

File: src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php [hunk 2]
```diff
@@ -71,8 +72,9 @@ public function processNode(Node $node, Scope $scope): array
             return [];
         }
 
-        return [
-            '\Drupal calls should be avoided in classes, use dependency injection instead'
-        ];
+        return
+            [RuleErrorBuilder::message('\Drupal calls should be avoided in classes, use dependency injection instead')
+             ->identifier('globalDrupalDependencyInjectio.useDependencyInjection')
```
Comment (mglaman) [inline src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php:77]:
```suggestion
             ->identifier('globalDrupalDependencyInjection.useDependencyInjection')
```

**Review by mglaman — COMMENTED** [review]
oh, did rule identifiers become required, essentially fixing https://github.com/mglaman/phpstan-drupal/pull/803 😱 I wish I could have landed that sooner to reduce this effort. I appreciate it. I literally fell asleep at my laptop 2 times while working on that.

**Comment by mglaman** [comment]
> A new branch might be a nice clean cut-off point to drop D9 support?

We can say 2.x drops Drupal 9 support, as a follow up to his PR.

File: src/Rules/Classes/PluginManagerInspectionRule.php (hunk 1 above)
Comment (Boegie) [reply src/Rules/Classes/PluginManagerInspectionRule.php:84]: Nope, this is a rare occasion where there was actually an identifier and I completely overlooked it.

Not too sure what to do now...

Do we want double-full-stop identifiers? 

If we do I'll have to change the other identifiers in this class to the same prefix.

Help! :)

**Comment by Boegie** [comment]
> I appreciate it. I literally fell asleep at my laptop 2 times while working on that.

I couldn't handle in batches longer than one hour, extremely grindy stuff

File: src/Rules/Classes/PluginManagerInspectionRule.php (hunk 1 above)
Comment (mglaman) [reply src/Rules/Classes/PluginManagerInspectionRule.php:84]: I'm fine taking your changes, I was just curious

**Comment by mglaman** [comment]
Thanks for this!!

**Pull request merged**